### ISSUE TITLE
Erroneous CursorStateError

### ISSUE THREAD

**Issue by kristopherdick-clc** [joowani/python-arango#78]
python-arango==4.2.0

If the collection has fewer documents than the batch size limit, the collection.all() rasies CursorStateError('cursor ID not set').

If the size of the result set is smaller than the batch size the lack of a cursor ID is expected and should, therefore, be ignored and the results returned.

```
File "/usr/local/lib/python3.6/site-packages/arango/cursor.py", line 292, in close
    raise CursorStateError('cursor ID not set')
arango.exceptions.CursorStateError: cursor ID not set
```

**Comment by joowani** [comment]
I released a fix in python-arango [4.2.1](https://github.com/joowani/python-arango/releases/tag/4.2.1). Please try it out and feel free to reopen this ticket if you have any other questions. Thanks again!

**Comment by kristopherdick-clc** [comment]
Confirmed. Thanks for getting to it so quickly.
